### PR TITLE
Feature/spring boot 2.x

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

**Comment by jwomeara** [comment]
The following PRs are all part of the feature/spring-boot-2.x effort and should be approved/merged together.

https://github.com/NationalSecurityAgency/datawave-accumulo-service/pull/8
https://github.com/NationalSecurityAgency/datawave-audit-service/pull/11
https://github.com/NationalSecurityAgency/datawave-authorization-service/pull/17
https://github.com/NationalSecurityAgency/datawave-config-service/pull/2
https://github.com/NationalSecurityAgency/datawave-dictionary-service/pull/17
https://github.com/NationalSecurityAgency/datawave-hazelcast-service/pull/3
https://github.com/NationalSecurityAgency/datawave-query-metric-service/pull/9
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-audit/pull/2
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-cache/pull/1
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter/pull/14
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-query-metric/pull/3
https://github.com/NationalSecurityAgency/datawave-in-memory-accumulo/pull/11
https://github.com/NationalSecurityAgency/datawave-accumulo-utils/pull/6
https://github.com/NationalSecurityAgency/datawave-base-rest-responses/pull/3
https://github.com/NationalSecurityAgency/datawave-common-utils/pull/2
https://github.com/NationalSecurityAgency/datawave-metadata-utils/pull/18
https://github.com/NationalSecurityAgency/datawave-metrics-reporter/pull/4
https://github.com/NationalSecurityAgency/datawave-type-utils/pull/18
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-metadata/pull/2
https://github.com/NationalSecurityAgency/datawave/pull/1876

File: api/src/main/java/datawave/webservice/model/FieldMapping.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public int compareTo(FieldMapping obj) {
+        
+        if (obj == null) {
+            throw new IllegalArgumentException("can not compare null");
+        }
+        
+        if (obj == this)
+            return 0;
+        
+        return new CompareToBuilder().append(datatype, ((FieldMapping) obj).datatype).append(fieldName, ((FieldMapping) obj).fieldName)
```
Comment (apmoriarty) [inline api/src/main/java/datawave/webservice/model/FieldMapping.java:46]: Seems odds to have to cast inside each append call. Maybe this way is cleaner nesting the CompareToBuilder inside an `instanceOf` check.

File: service/src/main/java/datawave/microservice/metadata/DefaultMetadataFieldScanner.java [hunk 2]
```diff
@@ -272,7 +271,7 @@ private void setType() {
         
         // Set the last updated date for the current {@link DefaultMetadataField} based on the timestamp of the current entry.
         private void setLastUpdated() {
-            currField.setLastUpdated(DATE_FORMATTER.print(currKey.getTimestamp()));
+            currField.setLastUpdated(new SimpleDateFormat(TIMESTAMP_FORMAT).format(new Date(currKey.getTimestamp())));
```
Comment (apmoriarty) [inline service/src/main/java/datawave/microservice/metadata/DefaultMetadataFieldScanner.java:274]: This might be a relatively expensive call

**Review by apmoriarty — DISMISSED** [review]
LGTM but someone with more spring experience should look it over

File: api/src/main/java/datawave/webservice/model/FieldMapping.java (hunk 1 above)
Comment (jwomeara) [reply api/src/main/java/datawave/webservice/model/FieldMapping.java:46]: Agreed.  This isn't new code - it was moved from here: https://github.com/NationalSecurityAgency/datawave/blob/integration/web-services/client/src/main/java/datawave/webservice/model/FieldMapping.java

Seems we've been doing it this way for some time.  I'm removing the casts in the spring-boot-2.x-accumulo-2.1 branch

File: service/src/main/java/datawave/microservice/metadata/DefaultMetadataFieldScanner.java (hunk 2 above)
Comment (jwomeara) [reply service/src/main/java/datawave/microservice/metadata/DefaultMetadataFieldScanner.java:274]: changing in the accumulo 2.1 branch

**Comment by jwomeara** [comment]
Closing this in favor of the spring-boot-2.x-accumulo-2.1 PR.

**Pull request closed without merging**